garage/np/algos/base.py

```diff
@@ -1,7 +1,95 @@
-class Algorithm:
-    pass
+from collections import deque
 
+import numpy as np
 
-class RLAlgorithm(Algorithm):
-    def train(self):
-        raise NotImplementedError
+from garage.logger import tabular
+from garage.misc import special
+from garage.tf.misc import tensor_utils
```

Comment:
This looks weird to me. Why do we need tf module dependency in `garage.np`?

Reply:
i think the root cause here is that the samplers should probably not be in `tf`

Reply:
It seems that the function used here is the same as in garage.misc.tensor_utils. I will update to use it to avoid tf dependency. Thanks for pointing out.